### ISSUE TITLE
'spack load' fails for packages that have been renamed

### ISSUE THREAD

**Issue by aweits** [spack/spack#14347]
Using 'spack load' on a package that was built prior to the renaming of that package will fail to load, even though the package is listed in 'spack find'.

```coltrane (builtin/packages):spack install -n testit
==> Installing testit
==> Searching for binary cache of testit
==> No binary for testit found: installing from source
==> Fetching file:///bin/true
######################################################################## 100.0%
==> Staging unexpanded archive /tmp/pytest-of-aweits/pytest-54/test_keep_exceptions0/tmp/spack-stage-testit-master-om6hvveukb7veteczuk4bfvppobpystk/true in /tmp/pytest-of-aweits/pytest-54/test_keep_exceptions0/tmp/spack-stage-testit-master-om6hvveukb7veteczuk4bfvppobpystk/spack-src
==> Created stage in /tmp/pytest-of-aweits/pytest-54/test_keep_exceptions0/tmp/spack-stage-testit-master-om6hvveukb7veteczuk4bfvppobpystk
==> No patches needed for testit
==> Building testit [Package]
==> Executing phase: 'install'
==> Successfully installed testit
  Fetch: 0.02s.  Build: 0.16s.  Total: 0.18s.
[+] /tmp/SPACKIT/spack/opt/spack/linux-centos7-ivybridge/gcc-4.8.5/testit-master-om6hvveukb7veteczuk4bfvppobpystk

[move and rename the package file]

coltrane (builtin/packages):spack find testit
==> 1 installed package
-- linux-centos7-ivybridge / gcc@4.8.5 --------------------------
testit@master
coltrane (builtin/packages):spack load testit
==> Error: Package 'testit' not found.
```

**Comment by hartzell** [comment]
There's a work in progress (#14062) that will change `spack load` so that it doesn't use modules; if so, this might become a non-problem.

If it is still useful, it might be worth printing out a warning/deprecation notice, although that might monkey up its use in dot files.

**Comment by aweits** [comment]
@hartzell #14062 does not currently address this issue (I just tested that branch over the weekend). I'm also relatively sure that #14062 will not preclude the use of modules, so this should be fixed anyways?

**Comment by alalazo** [comment]
Fixed by #14348
